### PR TITLE
Feature/mt 9/edit meal

### PR DESCRIPTION
**Pull request by kalibria**

### CODE REVIEW

File: src/list-of-meals/buttons/EditButton.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          minutes: timeManager.getMinutesUIFromTimeBl(mealBL),
+        })
+      );
+    });
+    dispatch(setNewMeal(true));
+  };
+
+  return (
+    <div>
+      {allMealsLength === 6 && (
```
Comment (ikushlianski) [inline src/list-of-meals/buttons/EditButton.tsx]: What does 6 mean? Looks like a magic number. Should either be a comment or a constant for it

File: src/list-of-meals/mealsManager.ts [hunk 2]
```diff
@@ -50,12 +50,34 @@ export class MealsManagerBL {
...
     if (isMealListInLS.length === 0) {
-      return this.getMealListBL();
+      return this.generateMealListBL();
     } else {
       return myLocalStorage.getMealListBL();
     }
   }
 
+  updateMealTime(mealList: IMealBL[], mealOrderNumEdited: number): IMealBL[] {
+    return mealList.reduce((acc: IMealBL[], item, index, _mealList) => {
```
Comment (ikushlianski) [inline src/list-of-meals/mealsManager.ts]: _mealList is not used so it should be removed to reduce confusion

File: src/settings/time.tsx [hunk 3]
```diff
@@ -46,6 +48,15 @@ export class Time implements TimeDate {
     }
   }
 
+  prepareHoursForModalWindow() {
+    const newHours: number[] = [];
+
+    for (let i = initialHourForModalWindow; i <= maxHourForModalWindow; ++i) {
```
Comment (ikushlianski) [inline src/settings/time.tsx]: try using `reduce` instead of `for` in such cases in the future

File: src/utility/validationMealTime.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { IMealBL } from '../list-of-meals/mealsManager';
+
+class ValidationMealTime {
+  isCurrentMealTimeLongerPrevious(
```
Comment (ikushlianski) [inline src/utility/validationMealTime.tsx]: could be also `isCurrMealLaterThanPrev`

**Review by ikushlianski — APPROVED** [review]

**Review by ikushlianski — CHANGES_REQUESTED** [review]
Magic number 6 should be removed

**Review by ikushlianski — APPROVED** [review]

File: src/list-of-meals/buttons/EditButton.tsx (hunk 1 above)
Comment (kalibria) [reply src/list-of-meals/buttons/EditButton.tsx]: deleted

File: src/list-of-meals/mealsManager.ts (hunk 2 above)
Comment (kalibria) [reply src/list-of-meals/mealsManager.ts]: removed

File: src/settings/time.tsx (hunk 3 above)
Comment (kalibria) [reply src/settings/time.tsx]: rewrote

File: src/utility/validationMealTime.tsx (hunk 4 above)
Comment (kalibria) [reply src/utility/validationMealTime.tsx]: changed name

**Review by ikushlianski — APPROVED** [review]

**Pull request merged**